### ISSUE TITLE
Selecting attributes with single character

### ISSUE THREAD

**Issue by AStoker** [dperini/nwmatcher#103]
Related to: https://github.com/tmpvar/jsdom/issues/1740

Trying to have a query that selects all the elements with the `t` attribute (used for localization). The regex used does not work with attributes that have a single character (trying to search for `tt` works fine). My query I was searching by was `[t]` (pretty basic), however, to test all the use cases (including the one in the wiki, since there isn't a "supported" query where you're just searching by attribute), I tried the `*[t]` to search every element with the `t` attribute. This also did not work correctly.  

It _appears_ that this line (https://github.com/dperini/nwmatcher/blob/master/src/nwmatcher-noqsa.js#L407) is the culprit. I'm not a master of Regex's, so I could be incorrect, but it appears to be trying to match the attributes that have _at least two_ characters. This could be due to having two identifiers (separated by the `:?`) in the regex. Having one identifier group would match to one character just fine, but when you provide the additional identifier group, you're ensuring that the matching text has at least 2 characters (two identifiers), since the `:` in there is optional, but the second identifier is not.  

Lots of words, yes. Not entirely sure how to solve it, except perhaps a quick regex change that makes the second identifier optional (allowing single character attributes), but I'm not sure what that would break.

**TL;DR**  
Single letter attributes aren't supported, e.g. `[t]`

**Comment by dperini** [comment]
Fixed issue tmpvar/jsdom#1740 here: https://github.com/dperini/nwmatcher/commit/7bd5c68

Thank you for the help and the suggestion !

**Comment by AStoker** [comment]
Awesome, I'll give it a quick test to confirm, but it looks good. Thanks for taking care of it!

**Comment by dperini** [comment]
The requested fix was released with version 1.4.0.

Thank you for the report !
